frex/utils/constraint_solver.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import annotations
+from ortools.linear_solver import pywraplp
+from frex.models import Candidate, ConstraintSectionSolution, ConstraintSolution
+from typing import Tuple
+from frex.utils.common import rgetattr
+from frex.utils import ConstraintType, Constraint
+from enum import Enum
+
+
+class ConstraintSolver:
+    """
+    A class to perform constraint solving to produce a final solution of items using constraints on the overall
+    set of items.
+    """
+
+    def __init__(self):
+        self.solver = pywraplp.Solver.CreateSolver("SCIP")
+        self.candidates = ()
```

Comment:
These can/should be private members, since they're only accessed by the fluent setters and `solve`.